src/VectorBundle.jl

```diff
@@ -0,0 +1,461 @@
+
+Compute the allocating variant of the [`FiberBundleProductRetraction`](@ref),
+which by default allocates and calls `retract_product!`.
+"""
+retract(::VectorBundle, p, q, t::Number, ::FiberBundleProductRetraction)
+
+function retract_product(M::VectorBundle, p, X, t::Number)
+    q = allocate_result(M, retract, p, X)
+    return retract_product!(M, q, p, X, t)
+end
+
+function retract_product!(B::VectorBundle, q, p, X, t::Number)
+    tX = t * X
+    xp, Xp = submanifold_components(B.manifold, p)
+    xq, Xq = submanifold_components(B.manifold, q)
+    VXM, VXF = submanifold_components(B.manifold, tX)
+    # this temporary avoids overwriting `p` when `q` and `p` occupy the same memory
+    xqt = exp(B.manifold, xp, VXM)
+    vector_transport_direction!(
+        B.manifold,
+        Xq,
+        xp,
+        Xp + VXF,
+        VXM,
+        B.vector_transport.method_point,
+    )
+    copyto!(B.manifold, xq, xqt)
+    return q
+end
+
+function retract_sasaki!(B::TangentBundle, q, p, X, t::Number, m::SasakiRetraction)
+    tX = t * X
+    xp, Xp = submanifold_components(B.manifold, p)
+    xq, Xq = submanifold_components(B.manifold, q)
+    VXM, VXF = submanifold_components(B.manifold, tX)
+    p_k = allocate(B.manifold, xp)
+    copyto!(B.manifold, p_k, xp)
+    X_k = allocate(B.manifold, Xp)
+    copyto!(B.manifold, X_k, Xp)
+    Y_k = allocate(B.manifold, VXM)
+    copyto!(B.manifold, Y_k, VXM)
+    Z_k = allocate(B.manifold, VXF)
+    copyto!(B.manifold, Z_k, VXF)
+    ϵ = 1 / m.L
+    for k in 1:(m.L)
+        p_kp1 = exp(B.manifold, p_k, ϵ * Y_k)
+        X_kp1 = parallel_transport_to(B.manifold, p_k, X_k .+ ϵ .* Z_k, p_kp1)
+        Y_kp1 = parallel_transport_to(
+            B.manifold,
+            p_k,
+            Y_k .+ ϵ .* riemann_tensor(B.manifold, p_k, X_k, Z_k, Y_k),
+            p_kp1,
+        )
+        Z_kp1 = parallel_transport_to(B.manifold, p_k, Z_k, p_kp1)
+        copyto!(B.manifold, p_k, p_kp1)
+        copyto!(B.manifold, X_k, p_kp1, X_kp1)
+        copyto!(B.manifold, Y_k, p_kp1, Y_kp1)
+        copyto!(B.manifold, Z_k, p_kp1, Z_kp1)
+    end
+    copyto!(B.manifold, xq, p_k)
+    copyto!(B.manifold, Xq, xq, X_k)
+    return q
+end
+
+function representation_size(B::TangentBundleFibers)
+    return representation_size(B.manifold)
+end
+
+function Base.show(io::IO, vb::VectorBundle)
+    return print(io, "VectorBundle($(vb.type.fiber), $(vb.manifold))")
+end
+function Base.show(io::IO, vbf::VectorBundleFibers)
+    return print(io, "VectorBundleFibers($(vbf.fiber.fiber), $(vbf.manifold))")
+end
+Base.show(io::IO, vb::TangentBundle) = print(io, "TangentBundle($(vb.manifold))")
+
+@inline function allocate_result(B::TangentBundleFibers, f::typeof(zero_vector), x...)
+    return allocate_result(B.manifold, f, x...)
+end
+
+"""
+    fiber_bundle_transport(fiber::FiberType, M::AbstractManifold)
+
+Determine the vector tranport used for [`exp`](@ref exp(::FiberBundle, ::Any...)) and
```

Comment:
this one I do not fully understand – how much is this different from a default_vector_transport_method? And why is the name without `_method`?